python_modules/dagster/dagster/_core/execution/plan/execute_step.py

```diff
@@ -121,6 +121,10 @@ def _process_user_event(
         for check_result in user_event.check_results or []:
             yield from _process_user_event(step_context, check_result)
 
+        # If a MaterializeResult was returned from an asset with no type annotation, the type will be
+        # interpreted as Any. We want to switch to Nothing, so that the I/O manager gets skipped
+        assets_def.op.output_dict[output_name]._dagster_type = Nothing  # noqa: SLF001
```

Comment:
eeee yikes - this is pretty sketchy. I am guessing this doesn't work for multiprocess execution since were just mutating in memory state.

Reply:
yeah it wouldn't ugh. ok so then maybe making `MaterializeResult` a recognized dagster type and then no coercing to `Nothing` is the right move